Lib/test/test_threading.py

```diff
@@ -2103,6 +2103,63 @@ def test__all__(self):
         support.check__all__(self, threading, ('threading', '_thread'),
                              extra=extra, not_exported=not_exported)
 
+    def test_set_name(self):
+        try:
+            get_name = _thread._get_name
+            _thread.set_name
+        except AttributeError:
+            self.skipTest("need thread._get_name() and thread.set_name()")
```

Comment:
Don't you prefer using `@unittest.skipUnless(hasattr(_thread, '_get_name'), "missing '_thread._get_name')` combined with `@unittest.skipUnless(hasattr(_thread, 'set_name'), "missing '_thread.set_name)`. At least, we'd know which one is missing (though you'll need `get_name = _thread._get_name` in the test)